Autotools.gitignore

```diff
@@ -41,3 +41,6 @@ m4/ltoptions.m4
 m4/ltsugar.m4
 m4/ltversion.m4
 m4/lt~obsolete.m4
+
+# Generated Makefile
```

Comment:
Can we make this comment a bit more descriptive? I worry about users pulling in templates and potentially not understanding _why_ an important file like this is actually safe to ignore.

Reply:
@shiftkey , I don't think it is necessary. Because, people who use Autotools are usually the people who know why.

In general, people who use those hand-write meta-build system are supposed to know how they work.

Reply:
Fine description added.